add unique identifier to messages

reggie3 opened the pull request:
Addresses issue #11

R1ZZU commented on WebView.js:
```diff
@@ -1,6 +1,9 @@
-import { WebView as NativeWebView } from 'react-native';
+import EventEmitter from "events";
+import React, { Component } from "react";
+import { WebView as NativeWebView } from "react-native";
```
I'd leave single quotes everywhere .This is common and well known part of javascript styleguide. Apart of that, every project has it's own styleguide as well all developers have there own coding style/preferences. If smth seems really weird to one, this should be raised as discussion point. Codestyle changes shouldn't be a part of feature pull request/bug fix

R1ZZU commented on WebView.js:
```diff
@@ -1,6 +1,9 @@
+import { WebView as NativeWebView } from "react-native";
+
+// package unique prefix for all messages
+const UNIQUE_MESSAGE_PREFIX = "f251c210-e7c9-42fa-bae3-b9352ec3722a";
```
not sure if this identifier should be SO unique 😄
One more point: this identifier should be shared between `react-native` and `WebView` so I'd better move it to separate file. Also `UNIQUE_MESSAGE_PREFIX` is a good name in general, but in that exact case I'd add more context to name to make it obvious that it relates to this package. `RN_MESSAGES_CHANNEL_PREFIX` works fine IMO

R1ZZU commented on WebView.js:
```diff
@@ -18,42 +21,58 @@ export class WebView extends Component {
     );
   }
 
-  _refWebView = (webview) => {
```
unjustified codestyle related changes (see first comment)

R1ZZU commented on WebView.js:
```diff
@@ -18,42 +21,58 @@ export class WebView extends Component {
-    const parsedMsg = JSON.parse(data);
+
+    if (data.indexOf(UNIQUE_MESSAGE_PREFIX) !== 0) {
+      return; // that's not something that was received from rn messages channel
```
this comment is redundant, as code is quite simple and self describing

R1ZZU commented:
Just to summarize: I see a lot of codestyle realted changes, so I think I should add eslint to avoid redundant changes

reggie3 commented:
I think my cod prettifier mist have changed the singles to doubles. As fir  the unique ID, I figured better safe than sorry, and it'd give you the option to change it. I'll see about making the changes you mentioned.

reggie3 commented:
I'm not sure if my fixes for the code review issues you brought up made it through.  Resubmitting a pull seems to work differently then doing it the first time.  I changed the double quotes to single, removed the code style issues that I was able to figure out, and created a separate config file to contain the unique ID.  I didn't shorten the ID though since I felt you'd probably have something specific in mind.

R1ZZU commented:
Now it looks really good! Thanks 👍 (will merge this one after verification everything works as expected)

reggie3 commented:
I have downloaded this package via NPM since this pull request was merged and the version that I receive does not appear to have received the update.  It doesn't include the config file or any use of it.

R1ZZU commented:
I've just published `v1.2.0` to npm, this version should include this fix

reggie3 commented:
Thanks. I thought I was missing something.

On Oct 9, 2017 4:05 PM, "Andrei Volchenko" <notifications@github.com> wrote:

> I've just published v1.2.0 to npm, this version should include this fix
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/R1ZZU/react-native-webview-messaging/pull/12#issuecomment-335271107>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHAb-JcduGaPc1Az0ccv_N0rdM4J4EeEks5sqnyIgaJpZM4PR2PB>
> .
>
